ProjFS.Mac/PrjFSLib/Json/JsonWriter.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <string>
+
+class JsonWriter
```

Comment:
@pmj @jeschu1 let me know your thoughts on where this class lives, and how I've done the unit tests.

With this PR things are setup as follows:

- There is a new `PrjFSLibTests` that tests the `PrjFSLib` dylib
- `JsonWriter.cpp`, though not required by `PrjFSLib`, is built as part of `PrjFSLib` (let me know what you think about this).
- I saw we're doing some weird things as far as directly compiling `.cpp` files from `PrjFSLib` in other projects (for example `PrjFSKextLogDaemon` builds `PrjFSUser.cpp` and now `JsonWriter.cpp` out of the `PrjFSLib` folder).  I did not change this as part of this PR.

Right now if there are no strong objections I'm leaning toward going with the current approach, and potentially having a follow up PR that cleans up `PrjFSLib` and how its files and binaries are being consumed.

Please let me know what you think and if you have any suggestions for cleaning this up.

Reply:
> * I saw we're doing some weird things as far as directly compiling `.cpp` files from `PrjFSLib` in other projects (for example `PrjFSKextLogDaemon` builds `PrjFSUser.cpp` and now `JsonWriter.cpp` out of the `PrjFSLib` folder).  I did not change this as part of this PR.

Yeah, I think if we're going to have multiple such shared utility compilation units, we should probably move them to their own directory, where the various targets can include them in their builds or not as needed. Then for unit testing them we can just have a test target that builds all of (and only) the shared compilation units.

Reply:
Filed #1182 for cleaning this up